### ISSUE TITLE
Documentation is lacking

### ISSUE THREAD

**Issue by mvasin** [fisshy/react-scroll#331]
Readme goes directly to code, without explaining what concepts the library is operating with; what is a Link, an Element, Events, animateScroll and so on and how it all fits together.

I think this is an issue. At least what is what and the basic mechanics / use cases should be explained in plain English.

**Comment by dimensi** [comment]
Update documentation please, i can't understand how work with scrollSpy and other objects from library.

**Comment by maracuja-juice** [comment]
It would also be great to know in the documentation which props for example will all be accepted.
When this is done it will be way easier to update the corresponding typescript typings.
